op-proposer/proposer/driver.go

```diff
@@ -68,42 +71,78 @@ type L2OutputSubmitter struct {
 
 	l2ooContract *bindings.L2OutputOracleCaller
 	l2ooABI      *abi.ABI
+
+	dgfContract *bindings.DisputeGameFactoryCaller
+	dgfABI      *abi.ABI
 }
 
 // NewL2OutputSubmitter creates a new L2 Output Submitter
 func NewL2OutputSubmitter(setup DriverSetup) (*L2OutputSubmitter, error) {
 	ctx, cancel := context.WithCancel(context.Background())
 
-	l2ooContract, err := bindings.NewL2OutputOracleCaller(setup.Cfg.L2OutputOracleAddr, setup.L1Client)
-	if err != nil {
-		cancel()
-		return nil, fmt.Errorf("failed to create L2OO at address %s: %w", setup.Cfg.L2OutputOracleAddr, err)
-	}
+	if setup.Cfg.DisputeGameFactoryAddr != nil {
+		dgfCaller, err := bindings.NewDisputeGameFactoryCaller(*setup.Cfg.DisputeGameFactoryAddr, setup.L1Client)
+		if err != nil {
+			cancel()
+			return nil, fmt.Errorf("failed to create DGF at address %s: %w", setup.Cfg.DisputeGameFactoryAddr, err)
+		}
 
-	cCtx, cCancel := context.WithTimeout(ctx, setup.Cfg.NetworkTimeout)
-	defer cCancel()
-	version, err := l2ooContract.Version(&bind.CallOpts{Context: cCtx})
-	if err != nil {
-		cancel()
-		return nil, err
-	}
-	log.Info("Connected to L2OutputOracle", "address", setup.Cfg.L2OutputOracleAddr, "version", version)
+		cCtx, cCancel := context.WithTimeout(ctx, setup.Cfg.NetworkTimeout)
+		defer cCancel()
+		version, err := dgfCaller.Version(&bind.CallOpts{Context: cCtx})
+		if err != nil {
+			cancel()
+			return nil, err
+		}
+		log.Info("Connected to DisputeGameFactory", "address", setup.Cfg.DisputeGameFactoryAddr, "version", version)
 
-	parsed, err := bindings.L2OutputOracleMetaData.GetAbi()
-	if err != nil {
-		cancel()
-		return nil, err
-	}
+		parsed, err := bindings.DisputeGameFactoryMetaData.GetAbi()
+		if err != nil {
+			cancel()
+			return nil, err
+		}
+
+		return &L2OutputSubmitter{
+			DriverSetup: setup,
+			done:        make(chan struct{}),
+			ctx:         ctx,
+			cancel:      cancel,
 
-	return &L2OutputSubmitter{
-		DriverSetup: setup,
-		done:        make(chan struct{}),
-		ctx:         ctx,
-		cancel:      cancel,
+			dgfContract: dgfCaller,
+			dgfABI:      parsed,
+		}, nil
+	} else {
```

Comment:
Instead of having two large indented branches, wdyt about having two ctors like `newDGFSubmitter` and `newL2OOSubmitter` and call them from here?